libraries/src/test/java/com/baeldung/mapdb/HelloBaeldungUnitTest.java

```diff
@@ -0,0 +1,51 @@
+package com.baeldung.mapdb;
+
+import org.junit.Test;
+import org.mapdb.DB;
+import org.mapdb.DBMaker;
+import org.mapdb.HTreeMap;
+
+import java.util.concurrent.ConcurrentMap;
+
+import static junit.framework.Assert.assertEquals;
+
+public class HelloBaeldungUnitTest {
+
+  @Test
+  public void givenInMemoryDBInstantiateCorrectly_whenDataSavedAndRetrieved_checkRetrievalCorrect() {
+
+    DB db = DBMaker.memoryDB().make();
+
+    final String WELCOME_MESSAGE_KEY = "Welcome Message";
+    final String WELCOME_MESSAGE = "Hello Baeldung!";
+
+    HTreeMap myMap = db.hashMap("myMap").createOrOpen();
+    myMap.put(WELCOME_MESSAGE_KEY, WELCOME_MESSAGE);
+
+    String welcomeMessage = (String) myMap.get(WELCOME_MESSAGE_KEY);
+
+    db.close();
+
+    assertEquals(WELCOME_MESSAGE, welcomeMessage);
+
+  }
+
+  @Test
+  public void givenInFileDBInstantiateCorrectly_whenDataSavedAndRetrieved_checkRetrievalCorrect() {
+
+    DB db = DBMaker.fileDB("file.db").make();
+
+    final String WELCOME_MESSAGE_KEY = "Welcome Message";
+    final String WELCOME_MESSAGE = "Hello Baeldung!";
+
+    HTreeMap myMap = db.hashMap("myMap").createOrOpen();
```

Comment:
Why does this not have specialised types? Can it only be done with no type specifier?

Reply:
It only accepts the specialised types when defining it alongside the `.keySerializer...` and `.valueSerializer` - which I didn't want to bring the serializers into the hello world example.